website/server/models/group.js

```diff
@@ -724,6 +724,10 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     pusher.trigger(`presence-group-${group._id}`, 'user-left', {
       userId: user._id,
     });
+
+    delete group.quest.members[user._id];
+    group.markModified('quest.members');
+    promises.push(group.save());
```

Comment:
It's a little weird to save the group and then possibly update or remove it below. I'm going to have to think on this a little.